cmd/gossamer/commands/import_state.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package commands
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/dot"
+	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	ImportStateCmd.Flags().String("chain", "", "Chain id used to load default configuration for specified chain")
+	ImportStateCmd.Flags().String("state-file", "", "Path to JSON file consisting of key-value pairs")
+	ImportStateCmd.Flags().String("header-file", "", "Path to JSON file of block header corresponding to the given state")
+	ImportStateCmd.Flags().Uint64("first-slot", 0, "The first BABE slot of the network")
+}
+
+// ImportStateCmd is the command to import a state from a JSON file
+var ImportStateCmd = &cobra.Command{
+	Use:   "import-state",
+	Short: "Import state from a JSON file and set it as the chain head state",
+	Long: `The import-state command allows a JSON file containing a given state
+in the form of key-value pairs to be imported.
+Input can be generated by using the RPC function state_getPairs.
+Example: 
+	gossamer import-state --state-file state.json --header-file header.json --first-slot <first slot of network>`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := execImportState(cmd); err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+func execImportState(cmd *cobra.Command) error {
```

Comment:
can we cover this in tests?

Reply:
I think it's kinda like a snapshot.  I wasn't aware we supported this as well.

Reply:
@kanishkatn is this functionality actually used anywhere?

Reply:
it's a sub command

Reply:
I've asked @dimartiro to cover this since he's working on the snapshot task.